proposals/NNNN-widget-support.md

```diff
@@ -0,0 +1,248 @@
+
+```xml
+<function name="Show" functionID="ShowID" messagetype="request" since="1.0">
+ :
+ :
+<param name="screenID" type="Integer" mandatory="false" since="5.1" />
+</function>
+
+<function name="SetDisplayLayout" functionID="SetDisplayLayoutID" messagetype="request" since="2.0">
+ :
+ :
+ <param name="screenID" type="Integer" mandatory="false" since="5.1" />
+</function>
+```
+
+This allows the app to use the full flexibility of the Show RPC in other screens.
+
+### Screen templates
+
+Today SDL comes with a set of templates available the app can choose from. The information what templates are available to the app is provided in 
+- `RegisterAppInterfaceResponse.displayCapabilities.templatesAvaiable` and 
+- `SetDisplayLayoutResponse.displayCapabilities.templatesAvailable`.
+
+Details about the currently selected screen template are part of the display capabilities struct:
+- text fields, the name, width etc.
+- image fields e.g. graphic, or soft button image resolutions
+
+The template design comes from the head unit. However predefined template names are defined in `PredefinedLayout` enum. The head unit should provide template designs for the predefined template names. Still it can also provide additional custom templates with self defined names. 
+
+All this can be reused for widget screens. When creating a new widget screen the available widget templates are provided in `CreateScreenResponse.displayCapabilities.templatesAvailable`.
+
+The details about the currently selected widget template are provided using the same display capabilities struct. It is expected that the HMI will return with fewer text fields (only main field 1 and 2), shorter text field width and smaller image resolutions. This proposed solution reuses the existing API to provide widget template details mentioned. Just as today the design for widgets also come from the head unit and the OEM.
+
+Examples for widget screens:
+
+| template name | Example layout |
+|-----------------------|----------------------------------------------------------------------------------------------|
+| text with graphic  | ![template](../assets/proposals/NNNN-widgets/template-text-with-graphic.png) |
+| graphic with text  | ![template](../assets/proposals/NNNN-widgets/template-graphic-with-text.png) |
+| tiles with graphic | ![template](../assets/proposals/NNNN-widgets/template-tiles-with-graphic.png) |
+
+### Screen types supported
+
+In order to inform the app what screen types are supported the struct `DisplayCapabilities` should be extended with a single optional parameter called `screenTypesSupported`.
+
+```xml
+<struct name="DisplayCapabilities" since="2.0">
+ :
+ <param name="screenTypesSupported" type="ScreenType" array="true" minsize="1" mandatory="false" />
+</struct>
+```
+
+This parameter does not inform the app how many screens of the type are supported but it informs if a display supports widgets or a main app screen. The HMI decides on 
```

Comment:
Should `the` be `this`?

This parameter does not inform the app how many screens of **the** type are supported but it informs if a display supports widgets or a main app screen. The HMI decides on

Reply:
changed it to *per*